File: backend/app/schemas/quote.py
from pydantic import BaseModel, Field
from typing import Optional, List
from decimal import Decimal
from datetime import datetime
from uuid import UUID

class DamageSelection(BaseModel):
    zone_id: str
    damage_type: str
    severity: int = Field(..., ge=1, le=5)

class QuoteRequest(BaseModel):
    vin: Optional[str] = Field(None, min_length=17, max_length=17)
    year: int = Field(..., ge=1900, le=2030)
    make: str = Field(..., min_length=1)
    model: str = Field(..., min_length=1)
    mileage: int = Field(..., ge=0)
    title_status: str = Field(..., pattern="^(clean|salvage|rebuilt|junk|lien)$")
    condition_rating: str = Field(..., pattern="^(excellent|good|fair|poor|junk)$")
    drivable: bool
    engine_issues: Optional[str] = None
    transmission_issues: Optional[str] = None
    exterior_damage: Optional[List[DamageSelection]] = None
    interior_damage: Optional[List[DamageSelection]] = None
    zip_code: str = Field(..., min_length=5, max_length=10)
    city: Optional[str] = None
    state: Optional[str] = None
    classification_hint: Optional[str] = None

class QuoteResponse(BaseModel):
    quote_id: UUID
    classification: str
    confidence: Decimal
    offer_amount: Optional[Decimal]
    offer_valid_until: datetime
    partner_id: Optional[int] = None
    calculation_method: str
    query_time_ms: float
    needs_human_review: bool = False

File: frontend/src/schemas/quote.ts
import { z } from 'zod'

export const damageSelectionSchema = z.object({
    zone_id: z.string(),
    damage_type: z.string(),
    severity: z.number().min(1).max(5),
})

export const quoteRequestSchema = z.object({
    vin: z.string().length(17).optional(),
    year: z.number().min(1900).max(2030),
    make: z.string().min(1),
    model: z.string().min(1),
    trim: z.string().optional(),
    mileage: z.number().min(0),
    title_status: z.enum(['clean', 'salvage', 'rebuilt', 'junk', 'lien']),
    condition_rating: z.enum(['excellent', 'good', 'fair', 'poor', 'junk']),
    drivable: z.boolean(),
    engine_issues: z.string().optional(),
    transmission_issues: z.string().optional(),
    exterior_damage: z.array(damageSelectionSchema).optional(),
    interior_damage: z.array(damageSelectionSchema).optional(),
    glass_damage: z.array(damageSelectionSchema).optional(),
    wheel_damage: z.array(damageSelectionSchema).optional(),
    zip_code: z.string().min(5).max(10),
    city: z.string().optional(),
    state: z.string().optional(),
    pickup_address: z.string().optional(),
    classification_hint: z.enum(['junk', 'auction']).optional(),
})

export type QuoteFormData = z.infer<typeof quoteRequestSchema>